lib/kafo/scenario_manager.rb

```diff
@@ -0,0 +1,123 @@
+      @config_dir = File.file?(config) ? File.dirname(config) : config
+      @last_scenario_link = File.join(config_dir, last_scenario_link_name)
+      @previous_scenario = File.realpath(last_scenario_link) if File.exists?(last_scenario_link)
+    end
+
+    def available_scenarios
+      # assume that *.yaml file in config_dir that has key :name is scenario definition
+      @available_scenarios ||= Dir.glob(File.join(config_dir, '*.yaml')).reject { |f| f =~ /#{last_scenario_link}$/ }.inject({}) do |scns, scn_file|
+        begin
+          content = YAML.load_file(scn_file)
+          scns[scn_file] = content if content.has_key?(:name)
+        rescue Psych::SyntaxError => e
+          warn "Warning: #{e}"
+        end
+        scns
+      end
+    end
+
+    def list_available_scenarios
+      say "Available scenarios"
+      available_scenarios.each do |config_file, content|
+        scenario = File.basename(config_file, '.yaml')
+        say "  #{content[:name]} (use: --scenario #{scenario})"
+        say "        " + content[:description] unless (content[:description].nil? || content[:description].empty?)
+      end
+      say "  No available scenarios found in #{config_dir}" if available_scenarios.empty?
+      KafoConfigure.exit(0)
+    end
+
+
+    def scenario_selection_wizard
+      wizard = KafoWizards.wizard(:cli, 'Select installation scenario',
+        :description => "Please select one of the pre-set installation scenarios. You can customize your installtion later during the installtion.")
+      f = wizard.factory
+      available_scenarios.keys.each do |scn|
+        wizard.entries << f.button(scn, :label => "#{available_scenarios[scn][:name]}: #{available_scenarios[scn][:description]}", :default => false)
+      end
+      wizard.entries << f.button(:cancel, :label => 'Cancel Installation', :default => false)
+      wizard
+    end
+
+    def select_scenario_interactively
+      # let the user select if in interactive mode
+      if (ARGV & ['--interactive', '-i']).any?
+        res = scenario_selection_wizard.run
+        if res == :cancel
+          say 'Installation was cancelled by user'
+          KafoConfigure.exit(0)
+        end
+        res
+      end
+    end
+
+    def scenario_changed?(scenario)
+      !!previous_scenario && scenario != previous_scenario
+    end
+
+    def configured?
+      defined?(CONFIG_DIR)
+    end
+
+    def select_scenario
+      scenario = scenario_from_args || previous_scenario ||
+        (available_scenarios.keys.count == 1 && available_scenarios.keys.first) ||
+        select_scenario_interactively
+      if scenario.nil?
+        KafoConfigure.logger.fatal  "Scenario was not selected, can not continue. Use --list-scenarios to list available options."
+        KafoConfigure.exit(:unknown_scenario)
+      else
+        # check if scenario was changed
+        confirm_scenario_change(scenario) if scenario_changed?(scenario)
```

Comment:
quite edge case but I hit it... if you run installer from directory in which you use some symlink it gets dereferenced for previous scenario because of https://github.com/theforeman/kafo/pull/117/files#diff-a7539345f28ead836346cbcefeb9debbR11 but the current scenario does not resulting in false change detection

I suppose we should wrap scenarion in `File.realpath($...)` as well